cvat-ui/src/components/export-dataset/export-dataset-modal.tsx

```diff
@@ -0,0 +1,142 @@
+        projects: { datasets: projectExportActivities, annotations: projectDumpActivities },
+    } = useSelector((state: CombinedState) => state.export);
+
+    const initActivities = (): void => {
+        if (instance instanceof core.classes.Project) {
+            setInstanceType('project');
+            setActivities(
+                (form.getFieldValue('saveImages') ? projectExportActivities : projectDumpActivities)[instance.id] || [],
+            );
+        } else if (instance instanceof core.classes.Task) {
+            setInstanceType('task');
+            setActivities(
+                (form.getFieldValue('saveImages') ? taskExportActivities : taskDumpActivities)[instance.id] || [],
+            );
+        }
+    };
+
+    useEffect(() => {
+        initActivities();
+    }, [instance?.id, instance instanceof core.classes.Project]);
+
+    const closeModal = (): void => {
+        form.resetFields();
+        dispatch(exportActions.closeExportModal());
+    };
+
+    const handleValuesChange = (changedValues: any): void => {
+        if ('saveImages' in changedValues) {
+            form.setFieldsValue({ selectedFormat: undefined });
+        }
+        initActivities();
+    };
+
+    const handleExport = (values: FormValues): void => {
+        // have to validate format before so it would not be undefined
+        dispatch(
+            exportDatasetAsync(instance, values.selectedFormat as string, values.customName || '', values.saveImages),
+        );
+        closeModal();
+    };
+
+    return (
+        <Form
+            name='Export dataset'
+            form={form}
+            labelCol={{ span: 8 }}
+            wrapperCol={{ span: 16 }}
+            onValuesChange={handleValuesChange}
+            initialValues={
+                {
+                    selectedFormat: undefined,
+                    saveImages: false,
+                    customName: undefined,
+                } as FormValues
+            }
+            onFinish={handleExport}
+        >
+            <Modal
+                title={`Export ${instanceType} #${instance?.id} as a dataset`}
+                visible={modalVisible}
+                onCancel={closeModal}
+                onOk={() => form.submit()}
+            >
+                <Form.Item
+                    name='selectedFormat'
+                    label='Export format'
+                    rules={[{ required: true, message: 'Format must be selected' }]}
+                >
+                    <Select placeholder='Select dataset format' className='cvat-modal-export-select'>
+                        {dumpers
+                            .sort((a: any, b: any) => a.name.localeCompare(b.name))
+                            .filter(
+                                (dumper: any): boolean =>
+                                    !(instance instanceof core.classes.Task && dumper.dimension !== instance.dimension),
```

Comment:
Difficult to understand.
Let's rewrite it as 

!(instance instanceof core.classes.Task) || dumper.dimension === instance.dimension